test-data/unit/check-overloading.test

```diff
@@ -1718,9 +1716,217 @@ def foo1(*args: int) -> int: ...  # E: Overloaded function signatures 1 and 2 ov
 def foo1(**kwargs: int) -> str: ...
 
 @overload
-def foo2(**kwargs: int) -> str: ...
+def foo2(**kwargs: int) -> str: ...  # E: Overloaded function signatures 1 and 2 overlap with incompatible return types
 @overload
-def foo2(*args: int) -> int: ...  # E: Overloaded function signature 2 will never be matched: function 1's parameter type(s) are the same or broader
+def foo2(*args: int) -> int: ...
```

Comment:
I am actually not so sure now we need to make this an error. This is very similar to empty list situation, maybe we can just pick the first overload?